### PR TITLE
User Profile Replacement/Removal

### LINKED ISSUE #9364: Redirect Profile Page to User Details, Remove Profile Component
User management redesign added a user details page. Most of the elements on this page are similar to what we have in profile page, i.e. logged in user would have options to edit their profile (including password change). 

Ensure all the components from profile are moved into this page for logged in user and remove the profile component.

### PR DESCRIPTION
**Pull request by Jacobjeevan**
## Proposed Changes

- Fixes #9364 

image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a new software update component for managing user software updates.
	- Added functionality for managing user information edits with improved error handling and notifications.
	- Enhanced user management interactions with a focus on new user creation.

- **Bug Fixes**
	- Adjusted verification processes in patient discharge tests to focus on essential elements.
	- Removed outdated user profile management tests and streamlined user management interactions.

- **Chores**
	- Removed unused components and imports related to user profile management to simplify the codebase.
	- Updated routing configuration to streamline user profile access.

### CODE REVIEW

File: src/Routers/routes/UserRoutes.tsx [hunk 1]
```diff
@@ -14,7 +13,7 @@ const UserRoutes: AppRoutes = {
   "/users/:username/:tab": ({ username, tab }) => (
     <UserHome username={username} tab={tab} />
   ),
-  "/user/profile": () => <UserProfile />,
+  "/user/:tab": ({ tab }) => <UserHome tab={tab} />,
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/Routers/routes/UserRoutes.tsx:16]: That is fine, in this case /profile leads to the page we want to display.

**Comment by nihal467** [comment]
@Jacobjeevan check the cypress failure

**Comment by Jacobjeevan** [comment]
> @Jacobjeevan check the cypress failure

Done

**Review by nihal467 — CHANGES_REQUESTED** [review]
- Evaluate both UserProfile.cy.ts and UserManage.cy.ts test files, identify common tests, keep one file, and delete the other.
- Remove any unused functions resulting from the above changes.
- Replace all submit button interactions with the reusable function cy.clickSubmitButton.
- Review the functionality of cy.typeIntoField() in our codebase to understand how a boolean parameter can be passed to clear the input field when required, eliminating the need for separate functions for clearing and inputting data.
- When creating a combination action (e.g., editUserDetails()), first create individual functions for each action and call them using this.functionName() within the combination function.
- Check the command.ts file and use reusable functions for all types of dropdowns and date selections.

**Comment by bodhish** [comment]
Why are we holding on to this>

File: src/components/Users/UserSummary.tsx [hunk 2]
```diff
@@ -200,12 +201,20 @@ export default function UserSummaryTab({
...
+              Child={LanguageSelector}
+              childProps={userColumnsData}
+            />
+            <UserColumns
+              heading={t("software_update")}
+              note={t("check_for_available_update")}
+              Child={UserSoftwareUpdate}
+              childProps={userColumnsData}
+            />
+          </>
```
[the remark this answers is not in the document]
Comment (rithviknishad) [reply src/components/Users/UserSummary.tsx:217]: Read the readme. Crowdin handles translations for languages other than English. We do not add them via PRs.

**Review by rithviknishad — APPROVED** [review]

File: cypress/e2e/users_spec/UsersManage.cy.ts [hunk 3]
```diff
@@ -50,45 +51,67 @@ describe("Manage User", () => {
...
-    manageUserPage.clickSubmit();
+    manageUserPage.clickUserInfoSubmitButton();
+    cy.verifyErrorMessages(contactInfoErrorMessages);
+    manageUserPage.editUserContactInfo(modifiedEmail, modifiedPhone);
+    manageUserPage.clickUserInfoSubmitButton();
+    manageUserPage.userInfoUpdateSuccessNotification();
     manageUserPage.clickContactInfoViewButton();
-    manageUserPage.verifyEditUserContactInfo("dev@gmail.com", "6234343435");
+    manageUserPage.verifyEditUserContactInfo(modifiedEmail, modifiedPhone);
   });
```
[the remark this answers is not in the document]
Comment (nihal467) [reply cypress/e2e/users_spec/UsersManage.cy.ts:110]: can be taken later

File: cypress/e2e/users_spec/UsersManage.cy.ts [hunk 4]
```diff
@@ -283,31 +312,36 @@ describe("Manage User", () => {
...
+      workinghour,
+      "https://www.example.com",
+    );
+    cy.clickSubmitButton();
+    cy.verifyNotification("User details updated successfully");
+    cy.closeNotification();
+    manageUserPage.verifyHoursAndVideoConnectLink(
+      workinghour,
+      "https://www.example.com",
+    );
```
[the remark this answers is not in the document]
Comment (nihal467) [reply cypress/e2e/users_spec/UsersManage.cy.ts:344]: can taken up later

**Comment by nihal467** [comment]
LGTM

**Review by nihal467 — APPROVED** [review]

**Pull request merged**